File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    /*
        User Params, inheriting from PaginationParams, will help filter Users in the main Discover users page
    */
    /// <summary>
    /// User Params, inheriting from PaginationParams, will help filter Users in the main Discover users page
    /// </summary>

    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";
        public string CyclingFrequency { get; set; }   // Daily, Weekly, Monthly
        public string CyclingCategory { get; set; } // Road, Gravel, Mountain
        public string SkillLevel { get; set; }    // Beginner, Intermediate, Advanced
        public string NameSearch { get; set; }
    }
}

File: client/src/app/_models/userParams.ts
import { User } from './user';

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 12;
    cyclingFrequency = 'all';
    cyclingCategory = 'all';
    skillLevel = 'all';
    nameSearch = '';
    state = '';
    city = '';
    orderBy = 'lastActive';

    constructor(user: User) {
        // userParams are for filtering user lists by the provided properties. If they don't provide male/female as a filter,
        // set gender to 'all' and server will recognize to not filter by gender.
        if (user.gender !== 'female' && user.gender !== 'male') {
            this.gender = 'all';
        }
    }
}
